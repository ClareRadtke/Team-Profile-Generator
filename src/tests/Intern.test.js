const { Intern } = require("../lib/classes");

describe(Intern, () => {
  it("should create a new Employee instance", () => {
    const intern = new Intern("John Doe", "123", "foo@example.com", "School");
    expect(intern).toMatchSnapshot();
  });
  describe("getName()", () => {
    it("should return the name", () => {
      const intern = new Intern("John Doe", "123", "foo@example.com", "School");
      expect(intern.getName()).toBe("John Doe");
    });
  });
  describe("getId()", () => {
    it("should return the id", () => {
      const intern = new Intern("John Doe", "123", "foo@example.com", "School");
      expect(intern.getId()).toBe("123");
    });
  });
  describe("getEmail()", () => {
    it("should return the email", () => {
      const intern = new Intern("John Doe", "123", "foo@example.com", "School");
      expect(intern.getEmail()).toBe("foo@example.com");
    });
  });
  describe("getRole()", () => {
    it("should return the role", () => {
      const intern = new Intern("John Doe", "123", "foo@example.com", "School");
      expect(intern.getRole()).toBe("Intern");
    });
  });
  describe("getSchool()", () => {
    it("should return the school", () => {
      const intern = new Intern("John Doe", "123", "foo@example.com", "School");
      expect(intern.getSchool()).toBe("School");
    });
  });
});
