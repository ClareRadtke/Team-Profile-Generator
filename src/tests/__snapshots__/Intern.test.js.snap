// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intern should create a new Employee instance 1`] = `
Intern {
  "email": "foo@example.com",
  "id": "123",
  "name": "John Doe",
  "school": "School",
}
`;
